### PR TITLE
provider: also get app from props.chidren

### PR DESCRIPTION
**Pull request by davibe**
addresses #2 but it's also backward compatible.

### CODE REVIEW

File: src/refnux.coffee [hunk 1]
```diff
@@ -89,8 +89,9 @@ class Provider extends Component
 
     constructor: (props) ->
         super
-        throw new Error("Provider does not support children") if props.children
+        throw new Error("Provider does not support multiple children") if props.children?.length > 1
         {@store, @app} = props
+        @app = props.children[0] if props.children?.length is 1
```
Comment (algesten) [inline src/refnux.coffee]: could we throw a big fat exception if `apps` is defined as prop _AND_ children?

File: test/test-provider.coffee [hunk 2]
```diff
@@ -27,7 +29,7 @@ describe 'Provider', ->
         eql app.args, [[]]
 
     it 'rerenders on store change', ->
-        pel = pf({app, store})
+        pel = pf({children: [app], store})
         renderToString pel
         store.dispatch -> panda:43
         eql app.args, [[],[]]
```
Comment (algesten) [inline test/test-provider.coffee:43]: and a test for that exception

**Review by algesten — CHANGES_REQUESTED** [review]

File: src/refnux.coffee (hunk 1 above)
Comment (davibe) [reply src/refnux.coffee]: I thought about it for 3 min then considered it stop energy.

Do you see a user making this mistake ? Does it have to fail badly ? What about a softer warning or nothing at all ?

File: test/test-provider.coffee (hunk 2 above)
Comment (davibe) [reply test/test-provider.coffee:43]: if you want, i'll do it.

Comment (davibe) [reply test/test-provider.coffee:43]: oh it's a change request. Ok i do it.

**Pull request merged**